### PR TITLE
Fix some mod interface discrepancies

### PR DESCRIPTION
**Pull request by peppy**
- Fix IApplicableMod being added at the wrong place in a few cases.
- Fix autoplay being selectable when it is actually only a dummy implementation.

### CODE REVIEW

File: osu.Game/Rulesets/Mods/ModAutoplay.cs [hunk 1]
```diff
@@ -5,30 +5,31 @@
...
         where T : HitObject
     {
-        protected abstract Score CreateReplayScore(Beatmap<T> beatmap);
+        protected virtual Score CreateReplayScore(Beatmap<T> beatmap) => new Score { Replay = new Replay() };
 
-        public virtual void ApplyToRulesetContainer(RulesetContainer<T> rulesetContainer)
-        {
-            rulesetContainer.SetReplay(CreateReplayScore(rulesetContainer.Beatmap)?.Replay);
-        }
+        public override bool HasImplementation => GetType().GenericTypeArguments.Length == 0;
```
Comment (smoogipoo) [inline osu.Game/Rulesets/Mods/ModAutoplay.cs]: I think this should be done at a base level, no? I foresee us having more generic mod implementations like this.

**Review by smoogipoo — CHANGES_REQUESTED** [review]

File: osu.Game/Rulesets/Mods/ModAutoplay.cs [hunk 2]
```diff
@@ -5,30 +5,31 @@
...
-    public class ModAutoplay : Mod, IApplicableFailOverride
+    public abstract class ModAutoplay : Mod
     {
         public override string Name => "Autoplay";
         public override string ShortenedName => "AT";
         public override FontAwesome Icon => FontAwesome.fa_osu_mod_auto;
         public override string Description => "Watch a perfect automated play through the song";
         public override double ScoreMultiplier => 0;
         public override Type[] IncompatibleMods => new[] { typeof(ModRelax), typeof(ModSuddenDeath), typeof(ModNoFail) };
-        public bool AllowFail => false;
```
Comment (Aergwyn) [inline osu.Game/Rulesets/Mods/ModAutoplay.cs]: Where did this go? 😕

Comment (peppy) [reply osu.Game/Rulesets/Mods/ModAutoplay.cs]: look up

File: osu.Game/Rulesets/Mods/ModAutoplay.cs (hunk 1 above)
Comment (peppy) [reply osu.Game/Rulesets/Mods/ModAutoplay.cs]: let's revisit this when another generic mod comes up

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**